Automattic/wp-calypso
# Checkout: add renewal gift route

#### Proposed Changes

This PR adds a new checkout route of the form `/checkout/:productSlugs/gift/:purchaseIds` to enable gifting subscription renewals to other WordPress.com users (see pbOQVh-2wh-p2 for more details). An example URL might be `/checkout/business-bundle-monthly,dotblog_domain/gift/10000,100001` to gift two products.

The new route works similarly to the current route for adding a renewal to the cart, which is `/checkout/:productSlugs/renew/:purchaseIds/:domain`. Both routes will attempt to add renewal products to the cart, but the gift route will allow adding a renewal for a _different user_. The user purchasing the gift does not need to have a WordPress.com site but does need to have an account.

Fixes https://github.com/Automattic/payments-shilling/issues/1194

(Most of this work was already prepared in the POC https://github.com/Automattic/wp-calypso/pull/69639 by @dylanrevisited)

#### Testing Instructions

You'll need two accounts and make sure that the store is sandboxed.

- With account A, purchase a personal plan (or use an account which already owns a plan).
- With account B, go to the following URL: `/checkout/personal-bundle/gift/<ACCOUNT_A_SUBSCRIPTION_ID>`. You can use Store Admin to find the subscription ID of the account A's plan.
- Verify that you see the gift subscription in account B's shopping cart.
